Customize timeout to avoid

**Is your feature request related to a problem? Please describe.**
I often have timeout error while running this job.

**Describe the solution you'd like**
Add a parameter to the action to customize timeout duration (default 10000ms).

**Additional context**

```
Error: Error while validating in Swagger Editor
Error: TimeoutError: Waiting failed: 10000ms exceeded
Error: Process completed with exit code 1.
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -83,3 +83,4 @@ jobs:
         with:
           definition-file: examples/openapi-2-0-error.yaml
           ignore-error: examples/ignore-error.js
+          default-timeout: 30000

--- a/README.md
+++ b/README.md
@@ -31,6 +31,11 @@ as a physical file in your repository.
 **Optional** Defines path to JavaScript file containing predicate for determining if the error should be ignored or not.
 
 
+### `default-timeout`
+
+**Optional** Defines maximum time in milliseconds a script waits for certain actions or events to occur.
+
+
 ## Example usage
 
 There are two major use-cases of how to use this GitHub Action.
@@ -87,4 +92,5 @@ jobs:
         with:
           swagger-editor-url: http://localhost/
           definition-file: examples/openapi-2-0.yaml
+          default-timeout: 20000
 ```

--- a/action.yml
+++ b/action.yml
@@ -12,12 +12,15 @@ inputs:
   ignore-error:
     description: JavaScript file containing predicate for determining if the error should be ignored or not
     required: false
+  default-timeout:
+    description: Maximum time in milliseconds a script waits for certain actions or events to occur
+    required: false
 runs:
   using: composite
   steps:
     - run: cd ${{ github.action_path }} && npm install
       shell: bash
-    - run: cd ${{ github.action_path }} && SWAGGER_EDITOR_URL=${{ inputs.swagger-editor-url }} DEFINITION_FILE=${{ inputs.definition-file }} IGNORE_ERROR=${{ inputs.ignore-error }} node src/index.js
+    - run: cd ${{ github.action_path }} && SWAGGER_EDITOR_URL=${{ inputs.swagger-editor-url }} DEFINITION_FILE=${{ inputs.definition-file }} IGNORE_ERROR=${{ inputs.ignore-error }} DEFAULT_TIMEOUT=${{ inputs.default-timeout }} node src/index.js
       shell: bash
 branding:
   icon: 'file-text'

--- a/src/index.js
+++ b/src/index.js
@@ -90,6 +90,7 @@ const defaultTimeout = parseInt(process.env.DEFAULT_TIMEOUT || '10000', 10);
 try {
   const definition = fs.readFileSync(definitionFilePath).toString();
 
+  page.setDefaultNavigationTimeout(defaultTimeout);
   page.setDefaultTimeout(defaultTimeout);
   await page.goto(process.env.SWAGGER_EDITOR_URL);
   await page.waitForSelector('.info .main .title', { visible: true });

--- a/src/index.js
+++ b/src/index.js
@@ -85,10 +85,12 @@ const definitionFilePath = path.join(
   process.env.GITHUB_WORKSPACE,
   process.env.DEFINITION_FILE
 );
+const defaultTimeout = parseInt(process.env.DEFAULT_TIMEOUT || '10000', 10);
 
 try {
   const definition = fs.readFileSync(definitionFilePath).toString();
 
+  page.setDefaultTimeout(defaultTimeout);
   await page.goto(process.env.SWAGGER_EDITOR_URL);
   await page.waitForSelector('.info .main .title', { visible: true });
   await page.waitForSelector('.ace_text-input', { visible: true });

--- a/src/index.js
+++ b/src/index.js
@@ -57,7 +57,7 @@ const parseErrors = async (page) => {
   try {
     await page.waitForSelector(
       '.swagger-ui .errors-wrapper .errors .error-wrapper',
-      { visible: true, timeout: 10000 }
+      { visible: true }
     );
   } catch {
     return errors;
@@ -106,7 +106,7 @@ try {
   await page.keyboard.up('Control');
   await page.waitForFunction(
     (text) => document.body.innerText.includes(text),
-    { timeout: 10000 },
+    {},
     'No API definition provided'
   );
   // paste in the OpenAPI description

